qiskit/visualization/circuit/matplotlib.py

```diff
@@ -441,6 +476,100 @@ def _get_layer_widths(self, node_data, glob_data):
+                        circuit_list.append(cases[0][1].copy_empty_like())
+                        for jump_values, circ in cases:
+                            node_data[node]["jump_values"].append(jump_values)
+                            circuit_list.append(circ)
+
+                    # Now process the circuits inside the ControlFlowOps
+                    for circ_num, circuit in enumerate(circuit_list):
+                        raw_gate_width = 0.0
+                        if circuit is None:  # If with no else
+                            self._flow_drawers[node].append(None)
+                            node_data[node]["width"].append(0.0)
+                            break
+
+                        # Depth of nested ControlFlowOp used for color of box
+                        if self._flow_parent is not None:
+                            node_data[node]["if_depth"] = (
+                                node_data[self._flow_parent]["if_depth"] + 1
+                            )
+                        # Update the wire_map with the qubits from the inner circuit
+                        inner_wire_map = {
+                            inner: wire_map[outer]
+                            for outer, inner in zip(self._qubits, circuit.qubits)
+                            if inner not in wire_map
+                        }
+                        wire_map.update(inner_wire_map)
+
+                        # Get the layered node lists and instantiate a new drawer class for
+                        # the circuit inside the ControlFlowOp.
+                        qubits, clbits, nodes = _get_layered_instructions(
+                            circuit, wire_map=wire_map
+                        )
+                        flow_drawer = MatplotlibDrawer(
+                            qubits,
+                            clbits,
+                            nodes,
+                            circuit,
+                            style=self._style,
+                            plot_barriers=self._plot_barriers,
+                            fold=self._fold,
+                            cregbundle=self._cregbundle,
+                        )
+                        self._flow_drawers[node].append(flow_drawer)
+
+                        # flow_parent is the parent of the new class instance
+                        flow_drawer._flow_parent = node
+
+                        # Recursively call _get_layer_widths for the circuit inside the ControlFlowOp
+                        flow_widths = flow_drawer._get_layer_widths(node_data, wire_map, glob_data)
+                        layer_widths.update(flow_widths)
+
+                        # Gates within a SwitchCaseOp need to know which case they are in
+                        for flow_layer in nodes:
+                            for flow_node in flow_layer:
+                                if isinstance(node.op, SwitchCaseOp):
+                                    node_data[flow_node]["case_num"] = circ_num
```

Comment:
How come they need to know?

Reply:
Check lines 699-704. They have to know which cases are to the left for width calcs.